### DISCUSSION TITLE
assigning empty array doesn't update the value.

### DISCUSSION THREAD

**Discussion by behzadmehrabi** [react-hook-form/react-hook-form#5922] [Q&A]
Hi, I'm using `fieldArray`  and let's say I have something like this:
```js
const items = Array(50).fill({
  name: "john",
  family: "",
  list: ["something"]
});
```
then I want to empty the `list` array, in useEffect. so I do something like this:
```js
React.useEffect(() => {
  fields.forEach((_field, i) => {
    update(i, {
      // assigning empty array doesn't work
      list: []
      // but assining filled array works.
      // list: ["hello from useEffect"]
    });
  });
}, []);
```
but the list doesn't get empty and still has a value of `["something"]`,
what is the problem?

codesandbox:
https://codesandbox.io/s/beautiful-rubin-2b8vg?file=/src/App.tsx:1038-1294

**Comment by behzadmehrabi** [comment]
@bluebill1049 any idea?

**Reply by bluebill1049** [reply]
Too busy... ping me again later, in case I miss it.
